scripts/mem_usage.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2014-2017, Linaro Limited
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Shows the memory usage '
+                                     'of a OP-TEE based on ELF sections')
+    parser.add_argument('tee_elf', help='the OP-TEE ELF file (tee.elf)')
+    parser.add_argument('-a', '--all', action='store_true',
+                        help=' same as -i -p -u -U')
+    parser.add_argument('-n', '--no-map', action='store_true',
+                        help=' do not show the detailed section mappings and '
+                        'RAM usage')
+    parser.add_argument('-i', '--init', action='store_true',
+                        help='report the total size of the .*_init sections')
+    parser.add_argument('-p', '--paged', action='store_true',
+                        help='report the total size of the .*_pageable '
+                        'sections')
+    parser.add_argument('-u', '--unpaged', action='store_true',
+                        help='report the total size of the unpaged sections, '
+                        'that is, all sections but the ones in --init or '
+                        '--paged')
+    parser.add_argument('-U', '--unpaged-no-heap', action='store_true',
+                        help='report the size of all unpaged sections '
+                        'excluding heap space. Reflects the size of unpaged '
+                        'code and data (.text, .rodata, .data, .bss, .nozi '
+                        'and possibly unwind tables)')
+    parser.add_argument('-r', '--raw', action='store_true',
+                        help='when processing -i, -p, -u, or -U, show only '
+                        'the size (in decimal) and no other text')
```

Comment:
Could we add the size of the pager pool: in kByte and pages.
I think we may need to produce a reference in the code to get it from the symbols in the ELF.